src/agent/src/sandbox.rs

```diff
@@ -40,6 +43,43 @@ pub const ERR_INVALID_CONTAINER_ID: &str = "Invalid container id";
 
 type UeventWatcher = (Box<dyn UeventMatcher>, oneshot::Sender<Uevent>);
 
+pub type StorageDeviceObject = Arc<Mutex<dyn StorageDevice>>;
+
+#[derive(Clone)]
+pub struct StorageState {
+    inner: StorageDeviceObject,
+}
+
+impl Debug for StorageState {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("StorageState").finish()
+    }
+}
+
+impl StorageState {
+    fn new() -> Self {
+        StorageState {
+            inner: Arc::new(Mutex::new(StorageDeviceGeneric::new("".to_string()))),
+        }
+    }
+
+    pub fn from_device(device: StorageDeviceObject) -> Self {
+        Self { inner: device }
+    }
+
+    pub async fn ref_count(&self) -> u32 {
+        self.inner.lock().await.ref_count()
```

Comment:
The mutex is only used for reference counter protection. I wonder if it is better to simply use an atomic value and remove the extra `StorageState` abstraction:
```
pub struct StorageDeviceObject {
    count: AtomicU32,
    inner: Arc<dyn StorageDevice>,
}
```

Reply:
Good point, simplified.